src/main/java/ru/rt/restream/reindexer/convert/GenericFieldConverter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Restream
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ru.rt.restream.reindexer.convert;
+
+import lombok.Getter;
+import ru.rt.restream.reindexer.convert.util.ResolvableType;
+import ru.rt.restream.reindexer.util.Pair;
+
+import java.util.Objects;
+
+/**
+ * A generic {@link FieldConverter} with resolved source and target types accessible via
+ * {@code GenericFieldConverter#getConvertiblePair}.
+ * @param <X> the field type used within the POJO
+ * @param <Y> the Reindexer stored database type
+ */
+public final class GenericFieldConverter<X, Y> implements FieldConverter<X, Y> {
```

Comment:
Is it necessary to implement FieldConverter? Maybe to register FieldConverters in the Registry, extend interface with the `Pair<X, Y> getConvertiblePair()` method or make a new one?

Reply:
I agree it can be transformed into an inner class within the factory to cache `Pair<ResolvableType, ResolvableType>` in it, I will also add a `getConvertiblePair` to the `FieldConverter` interface.